app/models/registration/campaign.rb

```diff
@@ -26,5 +26,17 @@ class Campaign < ApplicationRecord
                     completed: 3 }
 
     validates :title, :registration_deadline, :allocation_mode, :status, presence: true
+
+    def evaluate_policies_for(user, phase: :registration)
+      Registration::PolicyEngine.new(self).eligible?(user, phase: phase)
```

Comment:
I suggest something like this to avoid having to recreate a new object all the time:

```
def policy_engine
  @policy_engine ||= Registration::PolicyEngine.new(self)
end
```